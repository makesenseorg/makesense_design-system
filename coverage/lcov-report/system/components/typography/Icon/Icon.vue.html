<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for system/components/typography/Icon/Icon.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">system/components/typography/Icon</a> Icon.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;
  &lt;component
    :is="tag"
    :aria-label="ariaLabel"
    class="ds-icon"
  &gt;
    &lt;component
      v-if="svgComponent"
      :is="svgComponent"
      class="ds-icon-svg"/&gt;
  &lt;/component&gt;
&lt;/template&gt;
&nbsp;
&lt;script&gt;
import icons from '@@/icons'
/**
 * Icons are used to add meaning and improve accessibility.
 * @version 1.0.0
 */
export default {
  name: 'DsIcon',
  props: {
    /**
     * The name of the icon.
     */
    name: {
      type: String,
      required: true
    },
    /**
     * Descriptive text to be read to screenreaders.
     */
    ariaLabel: {
      type: String,
      default: 'icon'
    },
    /**
     * The html element name used for the icon.
     */
    tag: {
      type: String,
      default: 'span'
    }
  },
  computed: {
    svgComponent() {
      const icon = icons[this.name]
<span class="missing-if-branch" title="else path not taken" >E</span>      if (!icon) {
        return false
      }
<span class="cstat-no" title="statement not covered" >      return icon.render ? icon : icon.default</span>
    }
  }
}
&lt;/script&gt;
&nbsp;
&lt;style lang="scss" src="./style.scss"&gt;
&lt;/style&gt;
&nbsp;
&lt;docs src="./demo.md"&gt;&lt;/docs&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jan 22 2019 22:02:51 GMT+0100 (GMT+01:00)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
