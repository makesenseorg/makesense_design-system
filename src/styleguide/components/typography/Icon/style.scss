.ds-icon {
  @include reset;
  display: inline-flex;
  align-items: center;
  vertical-align: middle;
  height: 1em;
}

.ds-icon-svg {
  line-height: 1;
  height: 1.2em;
  // Use this if the icons are build with strokes
  // stroke: currentColor;
  // stroke-width: 2.5px;
  // stroke-linejoin: miter;
  // stroke-linecap: miter;
  // overflow: visible;
  // Use this if the icons are build with solids
  fill: currentColor
}