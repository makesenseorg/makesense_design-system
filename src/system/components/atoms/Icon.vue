<template>
  <font-awesome-icon
    v-if="library === 'font-awesome'"
    :icon="getFaIcon"
    :class="`icon icon--color-${color}`"
  />
  <svg
    v-else
    :class="`feather feather-${getIcon}`"
    xmlns="http://www.w3.org/2000/svg"
    :width="size"
    :height="size"
    :viewBox="`0 0 ${defaultSize} ${defaultSize}`"
    fill="none"
    stroke="currentColor"
    :stroke-width="strokeWidth"
    stroke-linecap="round"
    stroke-linejoin="round"
    v-html="path"
  ></svg>
</template>

<script>
// todo : import dynamically (without breaking use in nuxt.js)
import feather from "feather-icons";
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";

import {
  faArrowDown,
  faSort,
  faBars,
  faSearch,
  faArrowUp,
  faTimes,
  faTag,
  faArrowRight,
  faArrowLeft,
  faLock,
  faLifeRing,
  faBuilding,
  faCheck,
  faGlobeEurope,
  faClock,
  faLink,
  faWallet,
  faShare,
  faMapMarkerAlt
} from "@fortawesome/free-solid-svg-icons";

/**
 * Nice icons to make it more fun âœ¨
 * @version 0.1.0
 */
export default {
  name: "MksIcon",
  components: { FontAwesomeIcon },
  props: {
    /**
     * Type of icon depending on the library used. Prefer feather when possible
     */
    type: {
      type: String,
      default: ""
    },
    /**
     * Color of the icon amongs theme colors, inherits from parent by default.
     */
    color: {
      type: String,
      default: "inherit"
    },
    /**
     * Which library to use : "font-awesome", "feather"
     */
    library: {
      type: String,
      default: "feather"
    },
    size: {
      type: String,
      default: "24"
    },
    strokeWidth: {
      type: String,
      default: "2"
    }
  },
  data() {
    return {
      defaultSize: 24,
      path: ""
    };
  },
  async created() {
    if (this.library === "feather") {
      this.path = feather.icons[this.type];
    }
  },
  computed: {
    getIcon: function() {
      if (this.type === "close") return "x";
      if (this.type === "arrowUp") return "arrow-up";
      if (this.type === "arrowDown") return "arrow-down";
      if (this.type === "arrowLeft") return "arrow-left";
      if (this.type === "arrowRight") return "arrow-right";
      return this.type;
    },

    getFaIcon: function() {
      switch (this.type) {
        case "arrowDown":
          return faArrowDown;
        case "arrowUp":
          return faArrowUp;
        case "arrowRight":
          return faArrowRight;
        case "arrowLeft":
          return faArrowLeft;
        case "close":
          return faTimes;
        case "search":
          return faSearch;
        case "share":
          return faShare;
        case "check":
          return faCheck;
        case "sort":
          return faSort;
        case "link":
          return faLink;
        case "company":
          return faBuilding;
        case "lang":
          return faGlobeEurope;
        case "clock":
          return faClock;
        case "mapMarker":
          return faMapMarkerAlt;
        case "tag":
          return faTag;
        case "wallet":
          return faWallet;
        case "lock":
          return faLock;
        case "menu":
          return faBars;
        case "help":
          return faLifeRing;
        default:
          return faSearch;
      }
    }
  }
};
</script>

<style lang="scss" scoped>
.icon {
  color: inherit;
  vertical-align: middle;
  font-size: inherit;
  width: 1em;

  &--color-neutral {
    color: $color-neutral-70;
  }

  &--color-primary {
    color: $color-primary;
  }

  &--color-secondary {
    color: $color-secondary;
  }

  &--color-tertiary {
    color: $color-tertiary;
  }

  &--color-positive {
    color: $color-success;
  }

  &--color-warning {
    color: $color-warning;
  }

  &--color-negative {
    color: $color-danger;
  }
}
</style>
<docs>
## Feather icons (default)
List of all available icons : https://feathericons.com/

Some examples :

```jsx
<p>
  <mks-icon type="activity"></mks-icon> 
  <mks-icon type="alert-triangle"></mks-icon> 
  <mks-icon type="arrow-left"></mks-icon>
  <mks-icon type="arrow-left-circle"></mks-icon>
  <mks-icon type="bell"></mks-icon> 
  <mks-icon type="check"></mks-icon> 
  <mks-icon type="clock"></mks-icon> 
  <mks-icon type="eye-off"></mks-icon> 
  <mks-icon type="filter"></mks-icon> 
  <mks-icon type="heart"></mks-icon> 
  <mks-icon type="list"></mks-icon> 
  <mks-icon type="lock"></mks-icon>
  <mks-icon type="map-pin"></mks-icon>
  <mks-icon type="menu"></mks-icon>
  <mks-icon type="save"></mks-icon>
  <mks-icon type="star"></mks-icon> 
  <mks-icon type="thumbs-up"></mks-icon> 
  <mks-icon type="user"></mks-icon>
  <mks-icon type="zoom-in"></mks-icon>
</p>
```

## Font Awesome icons 
Font awesome icons are still available. However you have to specify to use the font-awesome library. 

Only these ones are available as of now : 

```jsx
<p>
  <mks-icon type="arrowUp" library="font-awesome"></mks-icon> 
  <mks-icon type="arrowDown" library="font-awesome"></mks-icon> 
  <mks-icon type="arrowLeft" library="font-awesome"></mks-icon>
  <mks-icon type="arrowRight" library="font-awesome"></mks-icon>
  <mks-icon type="close" library="font-awesome"></mks-icon> 
  <mks-icon type="mapMarker" library="font-awesome"></mks-icon> 
  <mks-icon type="lang" library="font-awesome"></mks-icon> 
  <mks-icon type="link" library="font-awesome"></mks-icon> 
  <mks-icon type="tag" library="font-awesome"></mks-icon> 
  <mks-icon type="company" library="font-awesome"></mks-icon> 
  <mks-icon type="search" library="font-awesome"></mks-icon> 
  <mks-icon type="check" library="font-awesome"></mks-icon>
  <mks-icon type="share" library="font-awesome"></mks-icon>
  <mks-icon type="sort"  library="font-awesome"></mks-icon>
  <mks-icon type="clock" library="font-awesome"></mks-icon>
  <mks-icon type="wallet" library="font-awesome"></mks-icon> 
  <mks-icon type="lock" library="font-awesome"></mks-icon> 
  <mks-icon type="menu" library="font-awesome"></mks-icon>
  <mks-icon type="help" library="font-awesome"></mks-icon>
</p>
```

## Colors 

Inherits from parent by default 

```jsx
<p>
  <mks-icon type="arrow-up" color="neutral"></mks-icon> 
  <mks-icon type="arrow-up" color="primary"></mks-icon> 
  <mks-icon type="arrow-up" color="secondary"></mks-icon> 
  <mks-icon type="arrow-up" color="tertiary"></mks-icon> 
  <mks-icon type="arrow-up" color="positive"></mks-icon> 
  <mks-icon type="arrow-up" color="warning"></mks-icon> 
  <mks-icon type="arrow-up" color="negative"></mks-icon> 
</p>
```
</docs>
