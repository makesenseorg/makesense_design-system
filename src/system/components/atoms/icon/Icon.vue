<template>
  <svg
    :class="`icon icon-${getIcon} icon--color-${color}`"
    role="img"
    xmlns="http://www.w3.org/2000/svg"
    :width="size"
    :height="size"
    :viewBox="`0 0 ${defaultSize} ${defaultSize}`"
    fill="none"
    :stroke-width="strokeWidth"
    stroke-linecap="round"
    stroke-linejoin="round"
    v-html="path"
  ></svg>
</template>

<script>
import feather from "feather-icons";
/**
 * Nice icons to make it more fun ✨ (based on feather)
 * @version 0.1.0
 */
export default {
  name: "MksIcon",
  props: {
    /**
     * Type of icon
     */
    type: {
      type: String,
      required: true
    },
    /**
     * Description of the icon for screen readers
     */
    alt: {
      type: String,
      required: false
    },
    /**
     * Color of the icon amongs theme colors, inherits from parent by default.
     */
    color: {
      type: String,
      default: "inherit",
      validator: function(value) {
        return (
          [
            "inherit",
            "primary",
            "secondary",
            "tertiary",
            "neutral",
            "positive",
            "warning",
            "negative",
            "currentColor"
          ].indexOf(value) !== -1
        );
      }
    },
    size: {
      type: [String, Number],
      default: "24"
    },
    strokeWidth: {
      type: String,
      default: "2"
    }
  },
  data() {
    return {
      defaultSize: 24
    };
  },
  computed: {
    getIcon: function() {
      if (this.type === "close") return "x";
      if (this.type === "arrowUp") return "arrow-up";
      if (this.type === "arrowDown") return "arrow-down";
      if (this.type === "arrowLeft") return "arrow-left";
      if (this.type === "arrowRight") return "arrow-right";
      return this.type;
    },
    path() {
      return `<title id="${this.type}">${this.alt || ""}</title>${
        feather.icons[this.type]
      }`;
    }
  }
};
</script>

<style lang="scss" scoped>
.icon {
  stroke: currentColor;
  vertical-align: middle;

  &--color-neutral {
    stroke: $color-neutral-70;
  }

  &--color-primary {
    stroke: $color-primary;
  }

  &--color-secondary {
    stroke: $color-secondary;
  }

  &--color-tertiary {
    stroke: $color-tertiary;
  }

  &--color-positive {
    stroke: $color-success;
  }

  &--color-warning {
    stroke: $color-warning;
  }

  &--color-negative {
    stroke: $color-danger;
  }
}
</style>
<docs>
## Icons
[List of 286 available icons](https://feathericons.com/)

Some examples :

```jsx
<p>
  <mks-icon type="activity" alt="Activités"></mks-icon> 
  <mks-icon type="alert-triangle" alt="Attention, il fait froid!"></mks-icon> 
  <mks-icon type="arrow-left" alt="A gauche"></mks-icon>
  <mks-icon type="arrow-left-circle" alt="Retourner"></mks-icon>
  <mks-icon type="bell" alt="Notifications"></mks-icon> 
  <mks-icon type="check"></mks-icon> 
  <mks-icon type="clock"></mks-icon> 
  <mks-icon type="eye-off"></mks-icon> 
  <mks-icon type="filter"></mks-icon> 
  <mks-icon type="heart"></mks-icon> 
  <mks-icon type="list"></mks-icon> 
  <mks-icon type="lock"></mks-icon>
  <mks-icon type="map-pin"></mks-icon>
  <mks-icon type="menu"></mks-icon>
  <mks-icon type="save"></mks-icon>
  <mks-icon type="star"></mks-icon> 
  <mks-icon type="thumbs-up"></mks-icon> 
  <mks-icon type="user"></mks-icon>
  <mks-icon type="zoom-in"></mks-icon>
</p>

<p>
  <mks-icon type="star" alt="petite étoile" v-bind:size="12" color="primary"></mks-icon> 
  <mks-icon type="star" alt="étoile couleur secondaire" size="50" color="secondary"></mks-icon> 
</p>
```
</docs>
