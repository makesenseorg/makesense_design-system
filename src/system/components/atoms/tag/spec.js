import {createLocalVue, shallowMount} from "@vue/test-utils";
import Comp from "./Tag.vue";
import Icon from "../icon/Icon.vue";

const localVue = createLocalVue();
localVue.component("MksIcon", Icon);

describe("Tag.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        label: "My Label"
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("matches snapshot without label", () => {
    const wrapper = shallowMount(Comp);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("matches snapshot with an icon", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      propsData: {
        label: "My Label",
        icon: "x"
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("truncates the label by default", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        label: "My very long label that should be truncated"
      }
    });
    expect(wrapper.text()).toBe("My very long label that should...");
  });

  it("keeps the label if trunc is false", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        label: "My very long label that should not be truncated",
        trunc: false
      }
    });
    expect(wrapper.text()).toBe(
      "My very long label that should not be truncated"
    );
  });

  it("has neutral as default color class", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        label: "My Label"
      }
    });
    expect(wrapper.classes()).toContain("tag--color-neutral");
  });

  it("sets the color class", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        label: "My Label",
        color: "primary"
      }
    });
    expect(wrapper.classes()).toContain("tag--color-primary");
  });

  it("throws on invalid color", () => {
    const color = Comp.props.color;
    expect(color.validator).toBeInstanceOf(Function);
    expect(color.validator("primary")).toBeTruthy();
    expect(color.validator("red")).toBeFalsy();
  });

  it("has medium as default size class", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        label: "My Label"
      }
    });
    expect(wrapper.classes()).toContain("tag--size-medium");
  });

  it("sets the size class", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        label: "My Label",
        size: "small"
      }
    });
    expect(wrapper.classes()).toContain("tag--size-small");
  });

  it("throws on invalid size", () => {
    const size = Comp.props.size;
    expect(size.validator).toBeInstanceOf(Function);
    expect(size.validator("small")).toBeTruthy();
    expect(size.validator("petit")).toBeFalsy();
  });

  it("is not-cliquable as default", () => {
    const wrapper = shallowMount(Comp);
    expect(wrapper.classes()).toContain("tag--not-cliquable");
  });

  it("sets the cliquable class", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        cliquable: true
      }
    });
    expect(wrapper.classes()).toContain("tag--cliquable");
  });

  it("emits click event", async () => {
    const wrapper = shallowMount(Comp);

    wrapper.trigger("click");
    await wrapper.vm.$nextTick();

    expect(wrapper.emitted().click).toBeTruthy();
  });

  it("throws on invalid icon", () => {
    const icon = Comp.props.icon;
    expect(icon.validator).toBeInstanceOf(Function);
    expect(icon.validator("x")).toBeTruthy();
    expect(icon.validator("menu")).toBeFalsy();
  });
});
