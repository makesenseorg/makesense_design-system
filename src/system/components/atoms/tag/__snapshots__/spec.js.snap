// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag.vue matches snapshot 1`] = `
<div
  class="tag tag--color-neutral tag--size-medium tag--not-cliquable"
>
  My Label 
  <!---->
</div>
`;

exports[`Tag.vue matches snapshot with an icon 1`] = `
<div
  class="tag tag--color-neutral tag--size-medium tag--not-cliquable"
>
  My Label 
  <mks-icon-stub
    class="tag__icon"
    color="inherit"
    library="feather"
    size="24"
    strokewidth="2"
    type="x"
  />
</div>
`;

exports[`Tag.vue matches snapshot without label 1`] = `
<div
  class="tag tag--color-neutral tag--size-medium tag--not-cliquable"
>
  -No label- 
  <!---->
</div>
`;
